### PR TITLE
Class Sorting + Websocket Fixes

### PR DESCRIPTION
**Pull request by Nxybi**

### CODE REVIEW

**Comment by Nicolas-Klatzer** [comment]
It should work now... hopefully, because I am not sure if changing the timeout solved the issue but it should work.

**Comment by Nxybi** [comment]
- [X] Webhook Fixed
- [ ] Extra Patching

**Comment by Nxybi** [comment]
fun fact: this PR is now 1 week old

**Comment by Nxybi** [comment]
oops

**Comment by Nicolas-Klatzer** [comment]
Custom Event_loop and proper loop closing work now. That means event_loops can be passed and then used and processed correctly. The connection can now also be closed correctly.

**Comment by Nicolas-Klatzer** [comment]
on_connection_start is now also triggered and on_init will now instead of returning the client object return the time when it started the connection

**Comment by Nicolas-Klatzer** [comment]
Only on_ready is still not called correctly. Will fix that tommorow

**Comment by Nxybi** [comment]
#1 and #2 are both fixed, maybe should merge?

File: openhivenpy/Types/House.py [hunk 1]
```diff
@@ -1,17 +1,78 @@
...
+        """
+        res = requests.post(f"https://api.hiven.io/v1/houses/{self._id}/rooms",headers={"Content-Type":"application/json","Authorization":self._TOKEN})
+        return res #ToDo
+
+    @staticmethod
+    def leave(self) -> bool:
+        """openhivenpy.Types.House.leave()
+
+        Leaves the House. Returns True if successful."""
+        res = requests.delete(f"https://api.hiven.io/v1/houses/{self._id}",headers={"Content-Type":"application/json","Authorization":self._TOKEN})
```
Comment (Nicolas-Klatzer) [inline openhivenpy/Types/House.py:77]: Should be replaced by HTTPClient and should request over that

**Comment by Nicolas-Klatzer** [comment]
I changed now the main logic. The connection to Hiven, which represents the client class and the websocket class, will directly be handled by the connection class. The events are now going to be handled specifically by a event handler that can be passed and modified

**Comment by Nicolas-Klatzer** [comment]
Resolved issue of the base websocket and split the functionality into different classes for better handling and also better overview of functionality. Merging now

**Comment by Nicolas-Klatzer** [comment]
The time has come to finally merge

**Pull request merged**